src/services/ledger/csv-wallet-export.ts

```diff
@@ -0,0 +1,70 @@
+import { LedgerService } from "@services/ledger"
+import { createObjectCsvStringifier, createObjectCsvWriter } from "csv-writer"
+
+const headers_field = [
+  "id",
+  "walletId",
+  "type",
+  "credit",
+  "debit",
+  "fee",
+  "currency",
+  "timestamp",
+  "pendingConfirmation",
+  "journalId",
+  "lnMemo",
+  "usd",
+  "feeUsd",
+  "recipientWalletId",
+  "username",
+  "memoFromPayer",
+  "paymentHash",
+  "pubkey",
+  "feeKnownInAdvance",
+  "address",
+  "txHash",
+]
+
+const header = headers_field.map((item) => ({ id: item, title: item }))
+
+export class CsvWalletsExport {
+  entries: LedgerTransaction[] = []
+
+  getBase64(): string {
+    const csvWriter = createObjectCsvStringifier({
+      header,
+    })
+
+    const header_stringify = csvWriter.getHeaderString()
+    const records = csvWriter.stringifyRecords(this.entries)
+
+    const str = header_stringify + records
+
+    // create buffer from string
+    const binaryData = Buffer.from(str, "utf8")
+
+    // decode buffer as base64
+    const base64Data = binaryData.toString("base64")
+
+    return base64Data
+  }
+
+  async saveToDisk(): Promise<void> {
+    const csvWriter = createObjectCsvWriter({
+      path: "export_accounts.csv",
+      header,
+    })
+
+    await csvWriter.writeRecords(this.entries)
+    console.log("saving complete")
+  }
+
+  async addWallet(wallet: WalletId): Promise<void | Error> {
+    // TODO: interface could be improved by returning self, so that it's
+    // possible to run csv.addWallet(wallet).getBase64()
+    const txs = await LedgerService().getLiabilityTransactions(wallet)
+    if (txs instanceof Error) return txs
```

Comment:
```suggestion
  async addWallet(walletId: WalletId): Promise<void | ApplicationError> {
    // TODO: interface could be improved by returning self, so that it's
    // possible to run csv.addWallet(wallet).getBase64()
    const txs = await LedgerService().getLiabilityTransactions(walletId)
    if (txs instanceof Error) return txs
```

Reply:
but `LedgerServiceError` is not a child of `ApplicationError`?

Reply:
It should be, included at this point: https://github.com/GaloyMoney/galoy/blob/709a3e1727bf101a70a3514f28a043612159e470/src/app/errors.ts#L17

Reply:
ok we use composition for `ApplicationError`. is it the only place we do it so?

Reply:
I think the easiest way to test that is to see if it's included in the `error-map.ts` switch (https://github.com/GaloyMoney/galoy/blob/709a3e1727bf101a70a3514f28a043612159e470/src/graphql/error-map.ts#L169)

That switch is designed to not compile if:
- something that isn't an `ApplicationError` gets included
- something that is an `ApplicationError` gets left out

Reply:
> ok we use composition for `ApplicationError`. is it the only place we do it so?

Yup only place where we don't use the `class ... extends .. ` pattern to create the error so far.